src/store/AuthContext.js

```diff
@@ -0,0 +1,188 @@
+import {useReducer,createContext,useState,useEffect,useContext} from 'react';
+import {warningToast,successToast,infoToast} from '../UI/Toast/Toast'
+import axios from '../useAxios'
+import {useHistory} from 'react-router-dom'
+
+export const AuthContext=createContext()
+
+export const useAuth=()=>useContext(AuthContext)
+
+export const AuthProvider=({children})=>{
+    const [loading,setLoading]=useState(false)
+    const {push}=useHistory();
+
+    const authManipulation=(state,action)=>{
+        switch (action.type) {
+            case "SIGNIN_USER":
+                return{
+                    ...state,
+                    userId:action.payload.userId,
+                    token:action.payload.token,
+                    userName:action.payload.userName,
+                    expiresIn:action.payload.expiresIn
+                }
+            case "SIGNOUT_USER":
+                return{
+                    ...state,
+                    userId:null,
+                    token:null,
+                    userName:null,
+                    expiresIn:null
+                }
+            default:
+                return state;
+        }
+    }
+
+    const [state,dispatch]=useReducer(authManipulation,{
+        userId:null,
+        token:null,
+        userName:null,
+        expiresIn:null,
+    })
```

Comment:
minor suggestion: U can extract the initial state to some variable , and also extract the reducer to another file --> Will make the Context file look cleaner !